tests/integrations/tests/utils/common.py

```
@@ -1865,4 +1866,410 @@ def get_content_string(content: Any) -> str:
+    Returns:
+        S3 URI string (e.g., s3://bucket/prefix/filename.jsonl)
+    """
+    import time
+    
+    if filename is None:
+        filename = f"batch-input-{int(time.time())}.jsonl"
+    
+    return f"s3://{bucket}/{prefix}{filename}"
+
+
+def assert_valid_batch_inline_response(response, provider: str = None) -> None:
```

Ruff (0.14.7)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)